include/sundials/impl/sundials_errors_impl.h

```diff
@@ -0,0 +1,183 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * !!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ * This is a 'private' header file and should not be used in user
+ * code. It is subject to change without warning.
+ * !!!!!!!!!!!!!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+ * -----------------------------------------------------------------
+ * Contains all error checking macros and private error handling API.
+ * -----------------------------------------------------------------*/
+
+#ifndef _SUNDIALS_ERRORS_IMPL_H
+#define _SUNDIALS_ERRORS_IMPL_H
+
+#include <sundials/sundials_errors.h>
+
+/* Shortcut macro to access the SUNContext declared with SUNAssignSUNCTX. */
+#define SUNCTX sunctx_
```

Comment:
I don't understand the utility/need for this macro, since it does nothing except change the case and underscore.  Was it more involved before, and was then simplified along the way?

Reply:
No I don't think it is that likely. Also the macro only really gets used by another macro with the final iteration of the `SUNCheck` macros which implicitly use the context. I prefer to keep it because it, along with `SUNAssignSUNCTX` ensure consistency and performance and allow us to tweak things later on without having to edit 200+ files.  

The purpose of `SUNAssignSUNCTX` is to allow us to have easy rule to follow on extracting `SUNContext` objects from arguments and using them in a way that mitigates any overhead.

Reply:
This is non-user-facing (an "impl.h" file), so that's not an issue.

I'm fine with keeping it, since it goes hand-in-hand with `SUNAssignSUNCTX`.

Reply:
This is an "impl" file but the macro does bleed into user-space when including `nvector_mpiplusx.h` and will occur with other headers as well if they're updated with inline functions.

Reply:
David is correct. It is not user-facing, but technically will be in the user-space because of inline functions. It is in a "private" header file that users should not access.